sources/kernel/interrupts.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <brutal/types.h>
+#include "kernel/domain.h"
+#include "kernel/task.h"
+
+typedef struct
+{
+    OBJECT_HEADER;
+    BrIrq irq;
+    BrIrqFlags flags;
+    Task *target_task;
+    Task *from_task;
+    BrMsg message;
+    bool ack;
+} IrqHandler;
```

Comment:
IMO this object should be split in two like:

```c
// KObject that can be passed across domains 
typedef struct
{
    OBJECT_HEADER;
    BrIrq irq;
} Irq;

// Binding between a task and an IRQ
typedef struct
{
    Irq* irq;
    Task *task;
    BrMsg msg;
    BrIrqFlags flags;
} IrqBinding;

typedef struct {
// ...
    Vec(IrqBinding) irq_bindings;
//...
} Task; 

// And BrIrqArgs become
typedef struct
{
    BrHandle irq;
    BrIrqFlags flags;
} BrIrqArgs;
```

This way when a task gets destroyed all the binding gets also destroyed

Reply:
But the bindings are destoyed with the domain ? + we still need The source thread for sending the message